consensus/src/rand/rand_gen/block_queue.rs

```diff
@@ -49,9 +49,11 @@ impl QueueItem {
         self.num_undecided_blocks
     }
 
-    pub fn rand_metadata(&self, round: Round) -> RandMetadata {
-        let block = self.blocks()[self.offset(round)].block();
-        block.into()
+    pub fn all_rand_metadata(&self) -> Vec<RandMetadata> {
```

Comment:
how about just make blocks(&self) public instead of construct this vector every time?